### PR TITLE
Feature/delete kafka topic

### PR DESCRIPTION
**Pull request by avnes**
Can delete on both blaster and confluent cloud.

### CODE REVIEW

**Review by ghost — DISMISSED** [review]
Hi Audun,

Of course Python is new to me so I cannot say I 100% understand all of the code, but at least what i've read seems to make some good sense.

I do, though, have some questions or suggestions.  Some of them might require Rasmus to provide some clarification on our standards though.

1. If I understand it right the readme is basically telling us how we can configure our environments to make use of the scripts.   However, the only play where we give a Windows-native method for installation is when we come to install Poetry.  I don't know if we should update the documentation to also describe how to do things like installing the Confluent CLI as well (there is a ZIP file you can pull from them and then unpack).  It depends on if we expect people to continue to work natively in Windows in the future or if we expect everyone to use Linux and/or WSL. 
2. It makes sense to have the sample_env file the way it is.  My guess is that it's preferred for security and once I have this file locally I should only need to populate it once (apart from the bearer token which needs updating more often).
3. Suggestion for v2 would maybe to add the option to provide a file name that contains the topics that want to be deleted.  It might be an easier way to work instead of having to edit the code file directly, but this is still a great starting point.

Other than that it looks good and gives me some great clues about how I need to structure my projects.  I'll use this in combination with your reference documentation to hopefully learn a bit more.

File: kafka-delete-topic/capability_service/kafka.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+
+class Topic:
+    """A class that represent a Kafka topic."""
+
+    def __init__(
+        self,
+        blaster_bearer_token: str,
+        blaster_topics_api_url: str,
+        blaster_cluster_uuid_prod: str,
```
Comment (wcarlsen) [inline kafka-delete-topic/capability_service/kafka.py]: Is it necessary to have dublicated variables for prod and dev, couldn't they just be one which differ dependent on environment?

File: kafka-delete-topic/capability_service/kafka.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        self.confluent_cluster_id_dev = confluent_cluster_id_dev
+        self.log_level: int = level
+
+    def _get_cluster_uuid(self, environment_name: str) -> str:
+        """
+        :param environment_name: A valid Kafka environment name. Currently only dev and prod clusters exist.
+        :type environment_name: str
+        :return: str
+        """
+        if environment_name.upper() in ["PROD", "PRODUCTION"]:
```
Comment (wcarlsen) [inline kafka-delete-topic/capability_service/kafka.py]: It seems like you are doing the same logic over and over again here caused by the dublicated variables above which is dependent on environment.

File: kafka-delete-topic/capability_service/kafka.py [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+        else:
+            raise InvalidKafkaCluster(environment_name)
+
+    def _get_confluent_environment_id(self, environment_name: str) -> str:
+        """
+        :param environment_name: A valid Confluence Kafka environment name. Currently only dev and prod clusters exist.
+        :type environment_name: str
+        :return: str
+        """
+        if environment_name.upper() in ["PROD", "PRODUCTION"]:
```
Comment (wcarlsen) [inline kafka-delete-topic/capability_service/kafka.py]: You are repeating logic here.

File: kafka-delete-topic/capability_service/kafka.py [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+        else:
+            raise InvalidKafkaCluster(environment_name)
+
+    def _get_confluent_cluster_id(self, environment_name: str) -> str:
+        """
+        :param environment_name: A valid Confluence Kafka environment name. Currently only dev and prod clusters exist.
+        :type environment_name: str
+        :return: str
+        """
+        if environment_name.upper() in ["PROD", "PRODUCTION"]:
```
Comment (wcarlsen) [inline kafka-delete-topic/capability_service/kafka.py]: Repeating logic again.

**Review by wcarlsen — CHANGES_REQUESTED** [review]
I don't like dublicated variables per environment. It shouldn't be needed. You repeat the same logic caused by the mulitple variables. I suggest you make a single config and load the config needed for the environment you are targeting.

Let me know if you have any questions.

File: kafka-delete-topic/capability_service/kafka.py (hunk 3 above)
Comment (avnes) [reply kafka-delete-topic/capability_service/kafka.py]: Fixed

File: kafka-delete-topic/capability_service/kafka.py (hunk 4 above)
Comment (avnes) [reply kafka-delete-topic/capability_service/kafka.py]: Fixed

File: kafka-delete-topic/capability_service/kafka.py (hunk 2 above)
Comment (avnes) [reply kafka-delete-topic/capability_service/kafka.py]: Now loading env variables from .env file through config class.

**Review by ghost — APPROVED** [review]
Looks good to me.

**Review by wcarlsen — APPROVED** [review]
LGTM!

**Pull request merged**